### PR TITLE
TRITON-2127 sdc-migrate fails for instances that use deletion_protection

### PR DESCRIPTION
**Pull request by twhiteman**

### CODE REVIEW

File: lib/backends/smartos/tasks/machine_migrate.js [hunk 1]
```diff
@@ -803,6 +803,76 @@ function setAutoboot(callback) {
 }
 
 
+function setIndestructibleZoneroot(callback) {
```
Comment (kusor) [inline lib/backends/smartos/tasks/machine_migrate.js:806]: Why this function signature if this callback is never called?

File: lib/backends/smartos/tasks/machine_migrate.js [hunk 2]
```diff
@@ -803,6 +803,76 @@ function setAutoboot(callback) {
...
+        log.warn({cmd: cmd, args: args}, 'Could not run vmadm update:', ex);
+        this.fatal(String(ex.stderr || buf));
+        return;
+    }
+
+    this.finish();
+}
+
+
+function setIndestructibleDelegated(callback) {
```
Comment (kusor) [inline lib/backends/smartos/tasks/machine_migrate.js:841]: Ditto, `callback` won't be used.

File: lib/backends/smartos/tasks/machine_migrate.js [hunk 3]
```diff
@@ -803,6 +803,76 @@ function setAutoboot(callback) {
...
+        log.warn({cmd: cmd, args: args}, 'Could not run vmadm update:', ex);
+        this.fatal(String(ex.stderr || buf));
+        return;
+    }
+
+    this.finish();
+}
+
+
 function start(callback) {
```
Comment (kusor) [inline lib/backends/smartos/tasks/machine_migrate.js:876]: The `start` function does not need a `callback`.  If you check recent tasks like `recovery_config` or `uninstall_agents` you'll see that those aren't using it at all. My bet is that it's just legacy convention from a past when it was needed.

File: lib/backends/smartos/tasks/machine_migrate.js (hunk 1 above)
Comment (twhiteman) [reply lib/backends/smartos/tasks/machine_migrate.js:806]: Copied from other examples I believe, but yes I've noticed that callback wasn't used and it bugged me - a good time to stop it :)

**Comment by twhiteman** [comment]
I've removed the unused migration callbacks and updated `lib/task_agent/tasks.js` as well.

**Pull request merged**